File: springboot-websocket-example/src/main/java/com/knowledgefactory/TextHandler.java
package com.knowledgefactory;

import java.io.IOException;

import org.json.JSONException;
import org.json.JSONObject;
import org.springframework.stereotype.Component;
import org.springframework.web.socket.TextMessage;
import org.springframework.web.socket.WebSocketSession;
import org.springframework.web.socket.handler.TextWebSocketHandler;

@Component
public class TextHandler extends TextWebSocketHandler {

	@Override
	public void handleTextMessage(WebSocketSession session, TextMessage message)
			throws InterruptedException, IOException, JSONException {

		String payload = message.getPayload();
		JSONObject jsonObject = new JSONObject(payload);
		session.sendMessage(new TextMessage("Your message:" + jsonObject.get("user")));
	}

}

File: kotlin-spring-main/kotlin_spring_websocket_example/src/main/kotlin/com/knf/dev/handler/TextHandler.kt
package com.knf.dev.handler

import org.springframework.web.socket.TextMessage
import org.json.JSONException
import org.json.JSONObject
import org.springframework.stereotype.Component
import java.io.IOException
import org.springframework.web.socket.WebSocketSession
import org.springframework.web.socket.handler.TextWebSocketHandler

@Component
class TextHandler : TextWebSocketHandler() {
    @Throws(InterruptedException::class, IOException::class, JSONException::class)
    public override fun handleTextMessage(session: WebSocketSession, message: TextMessage) {
        val payload = message.payload
        val jsonObject = JSONObject(payload)
        session.sendMessage(TextMessage("Your message:" + jsonObject.get("user")))
    }
}